includes/woopay/services/class-checkout-service.php

```diff
@@ -50,27 +50,30 @@ public function create_and_confirm_setup_intention_request( Request $base_reques
 		$request = WooPay_Create_And_Confirm_Setup_Intention::extend( $base_request );
 		$request->set_save_in_platform_account( $save_in_platform_account );
 		$request->set_save_payment_method_to_platform( $save_payment_method_to_platform );
-		$request->set_is_platform_payment_method( $this->is_platform_payment_method( $payment_information->is_using_saved_payment_method() ) );
+		$request->set_is_platform_payment_method( $this->is_platform_payment_method( $payment_information ) );
 		return $request;
 	}
 
 	/**
 	 * Determine if current payment method is a platform payment method.
 	 *
-	 * @param boolean $is_using_saved_payment_method If it is using saved payment method.
+	 * @param Payment_Information $payment_information Payment information object used to determine if a saved payment method is being used as well as helps to determine
+	 * if stripe platform account should be used or not.
 	 *
 	 * @return boolean True if it is a platform payment method.
 	 */
-	public function is_platform_payment_method( bool $is_using_saved_payment_method ) {
+	public function is_platform_payment_method( Payment_Information $payment_information ) {
 		// Return false for express checkout method.
 		if ( isset( $_POST['payment_request_type'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification
 			return false;
 		}
 
+		$should_use_stripe_platform = WC_Payments_Features::is_upe_deferred_intent_enabled() ? \WC_Payments::get_payment_gateway_by_id( $payment_information->get_payment_method_stripe_id() )->should_use_stripe_platform_on_checkout_page() : $this->should_use_stripe_platform_on_checkout_page();
```

Comment:
```suggestion
		$should_use_stripe_platform = WC_Payments_Features::is_upe_deferred_intent_enabled() ? \WC_Payments::get_payment_gateway_by_id( $payment_information->get_payment_method_stripe_id() )->should_use_stripe_platform_on_checkout_page() : \WC_Payments->get_gateway()->should_use_stripe_platform_on_checkout_page();
```

Just a thought, but when the deferred intent UPE is not enabled, why don't we summon the `should_use_stripe_platform_on_checkout_page()` function from `WC_Payments::get_gateway()` instead of using the one here? That way we could remove this method from the `Checkout_Service` class entirely and not have to duplicate this logic anymore? 🤔

Reply:
It definitely makes sense! Done in https://github.com/Automattic/woocommerce-payments/pull/6656/commits/8991d36285253931f956bb16c970c8063693994f